thinc/model.py

```diff
@@ -456,7 +456,9 @@ def get_gradients(self) -> Dict[Tuple[int, str], Tuple[FloatsXd, FloatsXd]]:
                 gradients[(node.id, name)] = (param, grad)
         return gradients
 
-    def copy(self: SelfT) -> SelfT:
+    def copy(
+        self: SelfT, copied_dict: Optional[Dict[int, Union["Model", Shim]]] = None
```

Comment:
`copied_dict` seems like a bit of a suboptimal name?

Also, I wonder if we should really expose this implementation detail in a public API. Maybe it would be better to rename this method to `_copy` and make a wrapper that does:

```
    def copy(self: SelfT) -> SelfT:
        return self._copy({})
```

Reply:
I've made this change and the code looks much better now.